commands/function/deploy.ts

```diff
@@ -88,11 +105,18 @@ exports.handler = async options => {
           functionPath,
         })
       );
-    } else if (e.status === 'ERROR') {
-      await outputBuildLog(e.cdnUrl);
+    } else if (
+      typeof e === 'object' &&
+      e !== null &&
+      'status' in e &&
+      e.status === 'ERROR' &&
+      'cdnUrl' in e &&
+      'errorReason' in e
+    ) {
+      await outputBuildLog(e.cdnUrl as string);
```

Comment:
These errors are tough to deal with - might be worth making a type + type guarding function for this so you dont have to cast stuff and do these checks inline

```
type FunctionBuildError = {
  status: 'ERROR';
  errorReason: string;
  cdnUrl: string;
};

function isFunctionBuildError(e: unknown): e is FunctionBuildError {
  return (
    typeof e === 'object' && e !== null && 'status' in e && e.status === 'ERROR'
  );
}```